src/time-picker/TimePicker.js

```diff
@@ -0,0 +1,140 @@
+  } else if (hour === 12) {
+    return `12${separator}${pad(minute)} pm`;
+  } else if (hour >= 13) {
+    return `${pad(hour % 12)}${separator}${pad(minute)} pm`;
+  } else {
+    return `${pad(hour)}${separator}${pad(minute)} am`;
+  }
+};
+
+const lteTime = (minTime, maxTime) => (
+  maxTime.hours > minTime.hours || (maxTime.hours === minTime.hours && maxTime.minutes > minTime.minutes)
+);
+
+const insertColon = str => {
+  if (numberRegex.test(str) && (str.length >= 3 && str.length <= 4)) {
+    const position = str.length - 2;
+    const strWithColon = `${str.substr(0, position)}${separator}${str.substr(position)}`;
+    return strWithColon;
+  }
+  return str;
+};
+const cleanString = str => str.replace(symbolRegex, ':');
+const cleanFilter = filterStr => insertColon(cleanString(filterStr));
+const filterOptions = (options, filterStr) => (
+  options.filter( option => (
+      option.value.substr(0, filterStr.length) === cleanFilter(filterStr) ||
+      option.label.substr(0, filterStr.length) === cleanFilter(filterStr) ||
+      option.label === cleanFilter(filterStr) ||
+      option.value === cleanFilter(filterStr)
+  ))
+);
+
+const Integer = t.refinement(t.Number, n => n % 1 === 0, 'Integer');
+const Hour = t.refinement(Integer, int => int >= 0 && int <= 23, 'Hour');
+const Minute = t.refinement(Integer, int => int >= 0 && int <= 59, 'Minute');
+const TimeFormat = t.enums.of([H12, H24], 'TimeFormat');
+const Time = t.struct({
+  hours: Hour,
+  minutes: Minute
+}, 'Time');
+
+const Props = t.refinement(t.struct({
+  onChange: t.Function,
+  value: t.maybe(Time),
+  minTime: t.maybe(Time),
+  maxTime: t.maybe(Time),
+  interval: t.maybe(Integer),
+  placeholder: t.maybe(t.String),
+  timeFormat: t.maybe(TimeFormat),
+  id: t.maybe(t.String),
+  className: t.maybe(t.String),
+  style: t.maybe(t.Object)
+}), ({ value, minTime, maxTime }) => (
+  lteTime(minTime, maxTime) && (!value || (lteTime(value, maxTime) && lteTime(minTime, value)))
+), 'Props');
+
+@skinnable()
+@props(Props)
+export default class TimePicker extends React.Component {
+
+  static defaultProps = {
+    placeholder: `--${separator}--`,
+    timeFormat: H24,
+    interval: 30,
+    minTime: { hours: 0, minutes: 0 },
+    maxTime: { hours: 23, minutes: 59 }
+  }
+
+  createOptionsList = ({ timeFormatter, minTime, maxTime, interval }) => {
+
+    const minHour = minTime.hours;
+    const maxHour = maxTime.hours;
+    const minMinute = minTime.minutes;
+    const maxMinute = maxTime.minutes;
+
+    const hours = range(minHour, maxHour + 1);
+    const minutes = range(0, 60, interval);
+    const options = flatten(hours.map(hour => minutes.filter(minute => (
+      (hour !== minHour || minute >= minMinute) &&
+      (hour !== maxHour || minute <= maxMinute)
+    )).map( minute => ({
+      value: `${hour}${separator}${minute}`,
+      label: timeFormatter(hour, minute)
+    })
+    )));
+    return options;
+  };
+
+  _onChange = (value) => {
+    if (value) {
```

Comment:
sorry, this is the `value` of `react-select` so it's called only when the user **clicks** on an option or if he clears it --> `value` can never be `''` but it can be `undefined`.

> Maybe I can implement an `else` statement that just call `this.props.onChange(value)` when the `value` is not well formatted.

yes, but when the `value` is `undefined` (which is falsy so it's ok a simple `else`)